packages/flutter/lib/src/rendering/table.dart

```diff
@@ -0,0 +1,875 @@
+
+  // Children are stored in row-major order.
+  // _children.length must be rows * columns
+  List<RenderBox> _children = const <RenderBox>[];
+
+  int get columns => _columns;
+  int _columns;
+  void set columns(int value) {
+    assert(value != null);
+    assert(value >= 0);
+    if (value == columns)
+      return;
+    int oldColumns = columns;
+    List<RenderBox> oldChildren = _children;
+    _columns = value;
+    _children = new List<RenderBox>()..length = columns * rows;
+    int columnsToCopy = math.min(columns, oldColumns);
+    for (int y = 0; y < rows; y += 1) {
+      for (int x = 0; x < columnsToCopy; x += 1)
+        _children[x + y * columns] = oldChildren[x + y * oldColumns];
+    }
+    markNeedsLayout();
+  }
+
+  int get rows => _rows;
+  int _rows;
+  void set rows(int value) {
+    assert(value != null);
+    assert(value >= 0);
+    if (value == rows)
+      return;
+    _rows = value;
+    _children.length = columns * rows;
+    markNeedsLayout();
+  }
+
+  Map<int, TableColumnWidth> _columnWidths;
+  void setColumnWidths(Map<int, TableColumnWidth> value) {
+    assert(value != null);
+    if (_columnWidths == value)
+      return;
+    _columnWidths = value;
+    markNeedsLayout();
+  }
+
+  void setColumnWidth(int column, TableColumnWidth value) {
+    if (_columnWidths[column] == value)
+      return;
+    _columnWidths[column] = value;
+    markNeedsLayout();
+  }
+
+  TableColumnWidth get defaultColumnWidth => _defaultColumnWidth;
+  TableColumnWidth _defaultColumnWidth;
+  void set defaultColumnWidth(TableColumnWidth value) {
+    assert(value != null);
+    if (defaultColumnWidth == value)
+      return;
+    _defaultColumnWidth = value;
+    markNeedsLayout();
+  }
+
+  TableBorder get border => _border;
+  TableBorder _border;
+  void set border(TableBorder value) {
+    if (border == value)
+      return;
+    _border = value;
+    markNeedsPaint();
+  }
+
+  TableCellVerticalAlignment get defaultVerticalAlignment => _defaultVerticalAlignment;
+  TableCellVerticalAlignment _defaultVerticalAlignment;
+  void set defaultVerticalAlignment (TableCellVerticalAlignment value) {
+    if (_defaultVerticalAlignment == value)
+      return;
+    _defaultVerticalAlignment = value;
+    markNeedsLayout();
+  }
+
+  TextBaseline get textBaseline => _textBaseline;
+  TextBaseline _textBaseline;
+  void set textBaseline (TextBaseline value) {
+    if (_textBaseline == value)
+      return;
+    _textBaseline = value;
+    markNeedsLayout();
+  }
+
+  @override
+  void setupParentData(RenderObject child) {
+    if (child.parentData is! TableCellParentData)
+      child.parentData = new TableCellParentData();
+  }
+
+  void setFlatChildren(int columns, List<RenderBox> cells) {
+    if (cells == _children && columns == _columns)
+      return;
+    for (RenderBox oldChild in cells) {
+      if (oldChild != null)
+        dropChild(oldChild);
```

Comment:
We'll probably eventually want a smarter version of this function that omits the drop/adopt pair if the identical child exists in the identical slot but we can wait for the widget layer to see exactly which of these functions is going to get used a lot.